sips/reduce_signatures_fork.md

```diff
@@ -0,0 +1,163 @@
+
+Moreover, upon a message creation, the cost of signing both with BLS and RSA keys, i.e. $530\mu$s + $570\mu$s, would be replaced by the cost of signing just with the RSA keys, i.e. $570\mu$s.
+
+
+## Spec change
+
+### Message structures
+
+We propose dropping both `SignedMessage` and `SignedPartialSignatureMessage` types and include the `FullData` field into the `Message` type as follows.
+
+```go
+type Message struct {
+	MsgType    MessageType
+	Height     Height
+	Round      Round
+	Identifier []byte `ssz-max:"56"`
+
+	Root                     [32]byte `ssz-size:"32"`
+	DataRound                Round
+	RoundChangeJustification [][]byte `ssz-max:"13,65536"`
+	PrepareJustification     [][]byte `ssz-max:"13,65536"`
+
+	FullData []byte `ssz-max:"5243144"` // New field
+}
+```
+
+To handle *decided* messages, we also propose allowing multiple signers and signatures in the `SignedSSVMessage`.
+
+
+```go
+type SignedSSVMessage struct {
+	OperatorID []OperatorID
+ 	Signature  [][]byte
+ 	Data       []byte
+}
+```
+
+### Decided messages
+
+With the proposed change, a *decided* message is identified by a `SignedSSVMessage` message with multiple signers and signatures.
+
+**Creation**: To create a *decided* message, no BLS aggregation is performed anymore. Now, the commit message data is sent under `SignedSSVMessage.Data` and the quorum (or more) of signers are listed in `SignedSSVMessage.OperatorID` along with their respective signatures in `SignedSSVMessage.Signature`.
+
+**Verification**: To verify a decided message, no more BLS verification is performed. Now, the number of signatures in `SignedSSVMessage.Signature` should be greater or equal to the cluster's quorum and the n-th signature should be correct for the public key of the n-th signer of `SignedSSVMessage.OperatorID` for $n = 0, 1, ..., len(Signatures) - 1$.
+
+
+### Consensus message processing
+
+The consensus module currently accepts `SignedMessage` messages. Now, it must accept `SignedSSVMessage` messages. `SignedSSVMessage` is required instead of the simple `Message` in order to allow the creation of messages with justifications.
+
+Except by messages inside justifications, the consensus module should not verify the signature of the received `SignedMessage` since it assumes that the message validation module already performed this check.
+
+Every sub-processing function of the consensus module, such as the `UponProposal`, `UponPrepare`, etc, should also receive a `SignedSSVMessage` instead of the deprecated `SignedMessage`.
+
+The message container must also stop storing `SignedMessage` objects in favor of storing `SignedSSVMessage` objects.
+
+### Partial signature message processing
+
+Currently, the partial signature processing functions accept a `SignedPartialSignatureMessage`. Now, it should accept a `SignedSSVMessage`. Similarly to the consensus module, it should assume that the message validation module already verified the RSA signature.
```

Comment:
latest changes are validating RSA signature inside the validator struct and not msg validation.